provider/pkg/provider/remote/commandOutputs.go

```diff
@@ -40,7 +40,10 @@ func (c *CommandOutputs) run(ctx p.Context, cmd string) error {
 
 	if c.Environment != nil {
 		for k, v := range *c.Environment {
-			session.Setenv(k, v)
+			err := session.Setenv(k, v)
+			if err != nil {
+				return fmt.Errorf("could not set environment variable '%s': %w. Please see the `environment` property's documentation for guidance", k, err)
```

Comment:
From a UX perspective, I would rather we split this up into two messages: a log message and an error:
```suggestion
				ctx.Logf(diag.Error, `Unable to set '%s'. This only works if your SSH server is configured to accept these variables via AcceptEnv. ternatively, if a Bash-like shell runs the command on the remote host, you could prefix the command itself with the variables in the form 'VAR=value command'`, k)
				return fmt.Errorf("could not set environment variable '%s': %w", k, err)
```

If you disagree, please feel free ignore this suggestion.

Reply:
fixed manually